# HG changeset patch
# User Lachlan Andrew <lachlanbis@gmail.com>
# Date 1453167066 -39600
#      Tue Jan 19 12:31:06 2016 +1100
# Node ID 4aa8e4b868c91b08f1598a978bca20435125abbc
# Parent  65827e9cccb87e602491d97bfa425214edc93285
Handle interrupt thrown by dbquit (bug #46931)

* oct-parse.in.yy (run): rethrow octave_interrupt_exception

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -4027,16 +4027,20 @@ octave_parser::run (void)
     {
       std::string file = lexer.fcn_file_full_name;
 
       if (file.empty ())
         error (e, "parse error");
       else
         error (e, "parse error in %s", file.c_str ());
     }
+  catch (octave_interrupt_exception &)
+    {
+      throw;
+    }
   catch (...)
     {
       std::string file = lexer.fcn_file_full_name;
 
       if (file.empty ())
         error ("unexpected exception while parsing input");
       else
         error ("unexpected exception while parsing %s", file.c_str ());
@@ -4083,16 +4087,20 @@ octave_push_parser::run (const std::stri
         {
           std::string file = lexer.fcn_file_full_name;
 
           if (file.empty ())
             error (e, "parse error");
           else
             error (e, "parse error in %s", file.c_str ());
         }
+      catch (octave_interrupt_exception &)
+        {
+          throw;
+        }
       catch (...)
         {
           std::string file = lexer.fcn_file_full_name;
 
           if (file.empty ())
             error ("unexpected exception while parsing input");
           else
             error ("unexpected exception while parsing %s", file.c_str ());
